staging/src/k8s.io/kubectl/pkg/polymorphichelpers/portsforobject.go

```diff
@@ -60,19 +61,21 @@ func portsForObject(object runtime.Object) ([]string, error) {
 }
 
 func getPorts(spec corev1.PodSpec) []string {
-	result := []string{}
+	result := sets.New[string]()
```

Comment:
Does this need to change as part of this PR, or can it be done separately?  I'm not saying it is wrong, but it is a subtle change in the output (if there are duplicate ports) and potentially the order of the output.  I'm not necessarily opposed to this change, just think that maybe it should be its own PR and would need some unit tests as well.

Reply:
I think this change is necessary because duplicate ports need to be eliminated. There should also be no external scheduling for this method. As for the unit test, I will add it to confirm whether there is a sequence problem.